### ISSUE TITLE
bbox_inside_weights, bbox_outside_weights

### ISSUE THREAD

**Issue by johncorring** [generalized-iou/Detectron.pytorch#1]
Hi, 
Thanks for sharing the code for this simple and interesting contribution. I have a conceptual question (not an actual issue) and I'd appreciate some insight.

I am able to understand the `compute_iou` function directly from the paper and it seems to correspond to the source, with the exception of the appearance of bbox_inside_weights and bbox_outside_weights in line 67 of lib/utils/net.py (and its use in 68,69).
Would you mind explaining this?

**Comment by jgwak** [comment]
In the original formulation of detectron, the final bounding box regression smooth l1 loss is `bbox_outside_weights * smooth_l1(bbox_inside_weights * bbox_diff)`. `bbox_inside_weights` sets the weight of positive examples to 1 and 0 otherwise. `bbox_outside_weights` is used to scale loss according to the number of positive samples. Furthermore, `bbox_outside_weights` is used to apply different scaling factor to different components of the bounding box loss, for example, multiplying `(5, 5, 10, 10)` to `(dx, dy, w, h)`. I applied `bbox_inside_weights` outside the loss function and took mean of component-wise scale factors because it makes less sense in IoU/GIoU which outputs a single value.

Closing the question. Please feel free to reopen it if you have any concerns or other questions
